src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1147,4 +1148,16 @@ public static boolean onThrowableImpact(EntityThrowable throwable, RayTraceResul
     {
         return MinecraftForge.EVENT_BUS.post(new ThrowableImpactEvent(throwable, ray));
     }
+
+    public static boolean onFarmlandTrample(World world, BlockPos pos, IBlockState target, EntityLivingBase entity)
+    {
+        BlockTrampleEvent event = new BlockTrampleEvent(world, pos, world.getBlockState(pos), target, entity);        
+        if (!MinecraftForge.EVENT_BUS.post(event))
+        {
+            world.setBlockState(pos, event.getTargetState());
+            return true;
```

Comment:
I was thinking convenience, so that someone who implemented a custom farmland who only cared wether the block changed could reuse this function. Not sure how likely that is, so Id appreciate some comments, considering it currently is "dead" code.